geode-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java

```diff
@@ -205,281 +98,27 @@ private boolean typeStoresDataInKeyRegistrar(RedisDataType type) {
-      } finally {
-        lock.unlock();
-      }
-    }
-  }
-
-  private Region<?, ?> getOrCreateRegion0(ByteArrayWrapper key, RedisDataType type,
-      ExecutionHandlerContext context, boolean addToMeta) {
-
-    keyRegistrar.validate(key, type);
-    Region<Object, Object> r = dynamicRegions.get(key);
-    if (r != null && r.isDestroyed()) {
-      removeKey(key, type);
-      r = null;
-    }
-    if (r == null) {
-      Lock lock = dynamicRegionLocks.get(key);
-      if (lock == null) {
-        Lock newLock = new ReentrantLock();
-        lock = dynamicRegionLocks.putIfAbsent(key, newLock);
-        if (lock == null) {
-          lock = newLock;
-        }
-      }
-
-      lock.lock();
-      try {
-        r = dynamicRegions.get(key);
-        if (r == null) {
-          boolean hasTransaction = context != null && context.hasTransaction(); // Can create
-          // without context
-          CacheTransactionManager txm = null;
-          TransactionId transactionId = null;
-          try {
-            if (hasTransaction) {
-              txm = cache.getCacheTransactionManager();
-              transactionId = txm.suspend();
-            }
-            Exception concurrentCreateDestroyException;
-            do {
-              concurrentCreateDestroyException = null;
-
-              r = createRegionGlobally(key.toString());
-
-              try {
-                if (type == RedisDataType.REDIS_LIST) {
-                  doInitializeList(key, r);
-                } else if (type == RedisDataType.REDIS_SORTEDSET) {
-                  try {
-                    doInitializeSortedSet(key, r);
-                  } catch (RegionNotFoundException | IndexInvalidException e) {
-                    concurrentCreateDestroyException = e;
-                  }
-                }
-              } catch (QueryInvalidException e) {
-                if (e.getCause() instanceof RegionNotFoundException) {
-                  concurrentCreateDestroyException = e;
-                }
-              }
-            } while (concurrentCreateDestroyException != null);
-            dynamicRegions.put(key, r);
-            if (addToMeta) {
-              keyRegistrar.register(key, type);
-            }
-          } finally {
-            if (hasTransaction) {
-              txm.resume(transactionId);
-            }
-          }
-        }
-      } finally {
-        lock.unlock();
-      }
-    }
-    return r;
-  }
-
-  /**
-   * SYNCHRONIZE EXTERNALLY OF this.locks.get(key)!!!!!
-   *
-   * @param key Key of region to destroy
-   * @param type Type of region to destroyu
-   * @return Flag if destroyed
-   */
-  private boolean destroyRegion(ByteArrayWrapper key, RedisDataType type) {
-    Region<?, ?> r = dynamicRegions.get(key);
-    if (r != null) {
-      try {
-        r.destroyRegion();
-      } catch (Exception e) {
-        return false;
-      } finally {
-        removeRegionState(key, type);
+      if (cancelExpiration) {
```

Comment:
So, because we don't care whether or not these calls return true or false, we don't really need the lock anymore? If two threads try to remove the same key, at least one will succeed so no harm, no foul?

Reply:
The locks on this method had to do with dynamic regions which no longer exist because they were only used by lists a zset. I do think we care about what we return from this and we end up being protected by the StripedExecutor now. I think once Strings have moved to the new data model this "del" code can be cleaned up some more. Also once everything is on the new expiration model then expiration cleanup this method does will not be needed.